src/AutoRest.CSharp/Common/Generation/LongRunningOperationWriter.cs

```diff
@@ -70,13 +70,21 @@ void WriteResultFunction(bool async)
             var @namespace = cs.Namespace;
             using (writer.Namespace(@namespace))
             {
+                bool isGenericOperation = !(operation.ResultType.IsFrameworkType && operation.ResultType.FrameworkType == typeof(Response));
                 writer.WriteXmlDocumentationSummary(operation.Description);
-                var interfaceType = new CSharpType(typeof(IOperationSource<>), operation.ResultType);
-                var baseType = new CSharpType(typeof(Operation<>), operation.ResultType);
-                var waitForCompletionType = new CSharpType(typeof(ValueTask<>), new CSharpType(typeof(Response<>), operation.ResultType));
-                var helperType = new CSharpType(typeof(OperationInternals<>), operation.ResultType);
-
-                using (writer.Scope($"{operation.Declaration.Accessibility} partial class {cs.Name}: {baseType}, {interfaceType}"))
+                var interfaceType = isGenericOperation ? new CSharpType(typeof(IOperationSource<>), operation.ResultType) : null;
+                var baseType = isGenericOperation ? new CSharpType(typeof(Operation<>), operation.ResultType) : new CSharpType(typeof(Operation));
+                var valueTaskType = isGenericOperation ? new CSharpType(typeof(Response<>), operation.ResultType) : new CSharpType(typeof(Response));
+                var waitForCompletionType = new CSharpType(typeof(ValueTask<>), valueTaskType);
+                var helperType = isGenericOperation ? new CSharpType(typeof(OperationInternals<>), operation.ResultType) : new CSharpType(typeof(OperationInternals));
+                var paramStart = isGenericOperation ? "(this, " : "(";
+                var waitForCompleteMethodName = isGenericOperation ? "WaitForCompletionAsync" : "WaitForCompletionResponseAsync";
+
+                FormattableString scope = isGenericOperation
+                    ? (FormattableString)$"{operation.Declaration.Accessibility} partial class {cs.Name}: {baseType}, {interfaceType}"
+                    : (FormattableString)$"{operation.Declaration.Accessibility} partial class {cs.Name}: {baseType}";
+
+                using (writer.Scope(scope))
```

Comment:
You can move the `$"{operation.Declaration.Accessibility} partial class {cs.Name}: {baseType}` before the scope call and have the logic inline.

``` C#
writer.Append("$"{operation.Declaration.Accessibility} partial class {cs.Name}: {baseType});
if (isGenericOperation)
{
	writer.Append(", interfaceType");
}
using (writer.Scope()) {}
```

Reply:
Basic idea is to have as little non-linearity in writing as possible.